src/Tasks/Copy.cs

```diff
@@ -361,6 +391,42 @@ private void LogAlwaysRetryDiagnosticFromResources(string messageResourceName, p
             return true;
         }
 
+        private void RunLsof()
+        {
+            try
+            {
+                using (Process process = new Process())
+                {
+                    process.StartInfo.FileName = "lsof";
+
+                    process.StartInfo.RedirectStandardOutput = true;
+                    process.StartInfo.RedirectStandardError = true;
+                    process.StartInfo.UseShellExecute = false;
+                    process.StartInfo.CreateNoWindow = true;
+
+                    process.Start();
+
+                    string output = process.StandardOutput.ReadToEnd();
+                    string error = process.StandardError.ReadToEnd();
+
+                    process.WaitForExit();
+                    if (!string.IsNullOrEmpty(output))
+                    {
+                        Log.LogMessage($"lsof output:\n{output}");
+                    }
+
+                    if (!string.IsNullOrEmpty(error))
+                    {
+                        Log.LogError($"lsof error:\n{error}");
+                    }
+                }
+            }
+            catch
+            {
+                Log.LogWarning("lsof invocation has failed.");
```

Comment:
I'm not convinced at the moment that failing in this kind of logging really deserves to take a no-error, no-warning case and turn it into an error/warning case...